_pytest/skipping.py

```diff
@@ -107,40 +116,51 @@ def _getglobals(self):
     def _istrue(self):
         if hasattr(self, 'result'):
             return self.result
-        if self.holder:
-            if self.holder.args or 'condition' in self.holder.kwargs:
-                self.result = False
-                # "holder" might be a MarkInfo or a MarkDecorator; only
-                # MarkInfo keeps track of all parameters it received in an
-                # _arglist attribute
-                marks = getattr(self.holder, '_marks', None) \
-                    or [self.holder.mark]
-                for _, args, kwargs in marks:
-                    if 'condition' in kwargs:
-                        args = (kwargs['condition'],)
-                    for expr in args:
+        self._marks = self._get_marks()
+
+        def needs_eval(mark):
+            return mark.args or 'condition' in mark.kwargs
+
+        if self._marks:
+            self.result = False
+            # "holder" might be a MarkInfo or a MarkDecorator; only
+            # MarkInfo keeps track of all parameters it received in an
+            # _arglist attribute
+            for mark in self._marks:
+                self._mark = mark
```

Comment:
`self._mark` will point to the last `mark` in `self._marks`, seems a little arbitrary to set this here. Is this by design? Otherwise it should be set in `self._get_marks()` instead to make it explicit that this is intended and doesn't depend on `_istrue` being called.

Reply:
it points to the mark that had the first true expression, this mess comes from merged mark and remembering epxressions, then using attributes from merged mrks


so if you had 2 marks with different expressions, and one had run=False, all would have run=False implied